espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/Element.java

```diff
@@ -68,46 +85,67 @@ public static ViewInteraction getViewInteractionById(String elementId) throws Ap
      * @param elementId
      * @return
      */
-    public static View getViewById(final String elementId) throws AppiumException {
-        UiControllerRunnable<View> runnable = new UiControllerRunnable<View>() {
-            @Override
-            public View run(UiController uiController) throws NoSuchElementException, StaleElementException {
-                if (!exists(elementId)) {
-                    throw new NoSuchElementException(String.format("Invalid element ID %s", elementId));
-                }
+    public static View getViewById(String elementId) throws AppiumException {
+        View view = getCachedView(elementId);
 
-                // Don't try accessing the View until the main thread is idle
-                uiController.loopMainThreadUntilIdle();
+        if (!view.isShown()) {
+            throw new StaleElementException(elementId);
+        }
+        return view;
+    }
 
-                // Get the cached view
-                View view = cache.get(elementId);
-                if (!view.isShown()) {
-                    throw new StaleElementException(elementId);
-                }
+    public static boolean exists(String elementId) {
+        return cache.containsKey(elementId);
+    }
 
-                return view;
-            }
-        };
+    public static View getCachedView(String elementId) throws NoSuchElementException, StaleElementException {
+        if (!exists(elementId)) {
+            throw new NoSuchElementException(String.format("No such element with ID %s", elementId));
+        }
 
-        return new UiControllerPerformer<>(runnable).run();
-    }
+        View view = cache.get(elementId);
 
-    public static View getViewById(String elementId, boolean skipWaitForMainThreadIdle) throws AppiumException {
-        if (skipWaitForMainThreadIdle) {
-            if (!exists(elementId)) {
-                throw new NoSuchElementException(String.format("Invalid element ID %s", elementId));
-            }
-            View view = cache.get(elementId);
+        if (!view.isShown()) {
+            throw new StaleElementException(elementId);
+        }
+
+        // This is a special case:
+        //
+        // If the contentDescription of a cached view changed, it almost certainly means the rendering of an
+        // AdapterView (ListView, GridView, Scroll, etc...) changed and the View contents were shuffled around.
+        //
+        // If we encounter this, try to scroll the element with the expected contentDescription into
+        // view and return that element
+        if (contentDescriptionCache.containsKey(elementId)) {
+            String expectedContentDesc = contentDescriptionCache.get(elementId);
+
+            if (view.getContentDescription() != null && !view.getContentDescription().equals(expectedContentDesc)) {
```

Comment:
!Objects.equals

Reply:
Does that work for String comparison though? The two strings being compared are different objects but I want to compare the values.